x/concentrated-liquidity/position.go

```diff
@@ -547,44 +552,69 @@ func (k Keeper) validatePositionsAndGetTotalLiquidity(ctx sdk.Context, owner sdk
 	return basePosition.PoolId, basePosition.LowerTick, basePosition.UpperTick, totalLiquidity, nil
 }
 
-// GetPositionIdToLock returns the positionId to lock mapping in state.
-func (k Keeper) GetPositionIdToLock(ctx sdk.Context, positionId uint64) (uint64, error) {
+// GetLockIdFromPositionId returns the lock id associated with the given position id.
+func (k Keeper) GetLockIdFromPositionId(ctx sdk.Context, positionId uint64) (uint64, error) {
 	store := ctx.KVStore(k.storeKey)
 
-	// Get the position ID to key mapping.
-	key := types.KeyPositionIdForLock(positionId)
-	value := store.Get(key)
+	// Get the position ID to lock ID mapping.
+	positionIdLockKey := types.KeyPositionIdForLock(positionId)
+	value := store.Get(positionIdLockKey)
 	if value == nil {
 		return 0, types.PositionIdToLockNotFoundError{PositionId: positionId}
 	}
 
 	return sdk.BigEndianToUint64(value), nil
 }
 
-// SetPositionIdToLock sets the positionId to lock mapping in state.
+// GetPositionIdToLockId returns the position id associated with the given lock id.
+func (k Keeper) GetPositionIdToLockId(ctx sdk.Context, underlyingLockId uint64) (uint64, error) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Get the lock ID to position ID mapping.
+	positionIdLockKey := types.KeyLockIdForPositionId(underlyingLockId)
+	value := store.Get(positionIdLockKey)
+	if value == nil {
+		return 0, types.LockIdToPositionIdNotFoundError{LockId: underlyingLockId}
+	}
+
+	return sdk.BigEndianToUint64(value), nil
+}
+
+// SetPositionIdToLock sets both the positionId to lock mapping and the lock to positionId mapping in state.
 func (k Keeper) SetPositionIdToLock(ctx sdk.Context, positionId, underlyingLockId uint64) {
 	store := ctx.KVStore(k.storeKey)
 
-	// Get the position ID to key mapping.
-	key := types.KeyPositionIdForLock(positionId)
-	store.Set(key, sdk.Uint64ToBigEndian(underlyingLockId))
+	// Get the position ID to key mapping and set.
+	positionIdLockKey := types.KeyPositionIdForLock(positionId)
+	store.Set(positionIdLockKey, sdk.Uint64ToBigEndian(underlyingLockId))
+
+	// Get the lock ID to key mapping and set.
+	lockIdKey := types.KeyLockIdForPositionId(underlyingLockId)
```

Comment:
can we add checks for this setter in the tests as well?

Reply:
This is tested in the CRUD test here https://github.com/osmosis-labs/osmosis/blob/04545e72285f1d99188fd7222667e71c650d2af3/x/concentrated-liquidity/position_test.go#L1146-L1157